server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1631,61 +1641,68 @@ public TreeMap<KeyExtent,TabletData> split(byte[] sp) throws IOException {
     Map<TabletFile,FileUtil.FileInfo> firstAndLastRows = FileUtil.tryToGetFirstAndLastRows(context,
         tableConfiguration, getDatafileManager().getFiles());
 
-    synchronized (this) {
-      // java needs tuples ...
-      TreeMap<KeyExtent,TabletData> newTablets = new TreeMap<>();
+    // This must be acquired before the tablet lock AND completeClose expects it to be acquired when
+    // called.
+    var lock = lockLogLock();
+    try {
+      synchronized (this) {
+        // java needs tuples ...
```

Comment:
There are Pair and Triple classes in commons-lang3. Immutable versions even.

Reply:
That comment was an existing comment, just made a white space change to that code.  Did not add the comment.